lib/si-rabbitmq/src/environment.rs

```diff
@@ -1,10 +1,14 @@
 use rabbitmq_stream_client::error::{StreamCreateError, StreamDeleteError};
 use rabbitmq_stream_client::types::{ByteCapacity, ResponseCode};
 use rabbitmq_stream_client::Environment as UpstreamEnvironment;
+use std::time::Duration;
+use telemetry::prelude::{info, trace, warn};
 
-use crate::{config::Config, error::RabbitResult};
+use crate::{config::Config, error::RabbitResult, RabbitError};
 
 const STREAM_LENGTH_CAPACTIY_IN_MEGABYTES: u64 = 10;
+const ENVIRONMENT_CREATION_TIMEOUT_DURATION_IN_SECONDS: u64 = 4;
+const ENVIRONMENT_CREATION_RETRIES: u64 = 3;
```

Comment:
If it's not ready in 12 seconds (3 retries x 4 second timeouts), then it is unlikely to be ready at all.